src/Sarif.Sarifer.Core/BackgroundAnalyzerBase.cs

```diff
@@ -57,7 +52,7 @@ public async Task<Stream> AnalyzeAsync(string path, string text, CancellationTok
         {
             text = text ?? throw new ArgumentNullException(nameof(text));
 
-            string solutionDirectory = await GetSolutionDirectoryAsync().ConfigureAwait(continueOnCapturedContext: false);
+            string solutionDirectory = await VsUtilities.GetSolutionDirectoryAsync().ConfigureAwait(continueOnCapturedContext: false);
```

Comment:
>VsUtilities [](http://example.com/codeflow?start=45&length=11)

I think we need to put this into its own leaf-node assembly (maybe the new Shell one) so all projects can access it.

Reply:
Started moving it to Shell project, but found there are some dependcies on Sarif.SDK and test flag etc need to be take care, it introduces many changes so I prefer to do that in another PR. created work item [https://github.com/microsoft/sarif-visualstudio-extension/issues/519](https://github.com/microsoft/sarif-visualstudio-extension/issues/519) for tracking.